kalite/topic_tools/content_recommendation.py

```diff
@@ -0,0 +1,506 @@
+'''
+All logic for generating and retriving content recommendations.
+'''
+
+from kalite.topic_tools import * 
+from kalite.main.models import ExerciseLog 
+
+
+###
+# The KING function for content recommendation. Call this single function for all 
+# of your content recommendation needs (i.e. in the API call).
+#
+# @param user: the unique user value of the current user (id/name I believe?)
+# @param current_subtopic: optional latest subtopic id that the user has worked on/completed. This
+#                          will act as a starting point for some of the algorithms (nearest neighbor, etc.)
+#
+# @return: Still deciding on this, but I assume that it will be a dictionary in the 
+#          form: { 'next_steps':[ ... ], 'explore':[ ... ], etc. } 
+###
+def get_recommendations(user=None, current_subtopic=None):
+
+    #the recommendations to return
+    result = {
+        "resume":{},        #not yet completed exercises, but have been started
+        "next_steps":{},    #things user is struggling on, next relevant items
+        "explore":{}        #less relevant exercises, for exploration
+    }
+
+    #preliminary user check
+    if not user:
+        #return 'Whoa there buddy, you did not specify a user!'
+        user = 'Yamira Jones (Facility: Wilson Elementary (#1829))'
```

Comment:
Looks good! I'm just wrapping up some of the API work tonight. Can we set
up some time Thursday or Friday to reconnect? Anytime before 11am PT

Al



On Monday, April 27, 2015, almgong <notifications@github.com> wrote:

> In kalite/topic_tools/content_recommendation.py
> <https://github.com/learningequality/ka-lite/pull/3549#discussion_r29216592>
> :
>
> > +# @return: Still deciding on this, but I assume that it will be a dictionary in the
> > +#          form: { 'next_steps':[ ... ], 'explore':[ ... ], etc. }
> > +###
> > +def get_recommendations(user=None, current_subtopic=None):
> > +
> > +    #the recommendations to return
> > +    result = {
> > +        "resume":{},        #not yet completed exercises, but have been started
> > +        "next_steps":{},    #things user is struggling on, next relevant items
> > +        "explore":{}        #less relevant exercises, for exploration
> > +    }
> > +
> > +    #preliminary user check
> > +    if not user:
> > +        #return 'Whoa there buddy, you did not specify a user!'
> > +        user = 'Yamira Jones (Facility: Wilson Elementary (#1829))'
>
> OK - using FacilityUser model as input
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/learningequality/ka-lite/pull/3549/files#r29216592>.
>